### PR TITLE
cxx-qt-gen: inject the CXX definition for common methods in writer

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Requires #206

### CODE REVIEW

**Comment by ahayzen-kdab** [comment]
Lets wait for the namespace one to land first as it's more important for other branches to rebase onto.

File: cxx-qt-gen/src/writer/rust/mod.rs [hunk 1]
```diff
@@ -18,9 +66,18 @@ pub fn write_rust(generated: &GeneratedRustBlocks) -> TokenStream {
...
 
     // Retrieve the module contents and namespace
-    let cxx_mod = &generated.cxx_mod;
+    let mut cxx_mod = generated.cxx_mod.clone();
     let cxx_qt_mod_contents = &generated.cxx_qt_mod_contents;
     let namespace = &generated.namespace;
+    let namespace_internals = &generated.namespace_internals;
+
+    // Inject the common methods into the bridge which we need
+    write_cxx_common_methods(
```
Comment (LeonMatthesKDAB) [inline cxx-qt-gen/src/writer/rust/mod.rs:75]: I feel like this shouldn't be handled by passing a mutable reference to the cxx_mod.
It leads to code duplication in terms of `content.push(syn::parse2(...)).expect("...")`.

Maybe refactor write_cxx_common_methods to:
`cxx_common_blocks` that returns a `Vec<TokenStream>`.
Then just loop over that:
``` rust
let cxx_mod_items = cxx_mod.content.as_mut().expect("").1;
for block in cxx_common_blocks(cpp_struct_ident, rust_struct_ident, namespace_internals){ 
    cxx_mod_items.push(syn::parse2(block).expect("Could not build extern block!"));
}
```

File: cxx-qt-gen/src/writer/rust/mod.rs [hunk 2]
```diff
@@ -6,6 +6,54 @@
...
+/// Add common methods into the CXX bridge which the C++ writer adds as well
+fn write_cxx_common_methods(
+    content: &mut Vec<Item>,
+    cpp_struct_ident: &Ident,
+    rust_struct_ident: &Ident,
+    namespace_internals: &String,
+) {
+    content.push(
+        syn::parse2(quote! {
+            unsafe extern "C++" {
```
Comment (LeonMatthesKDAB) [inline cxx-qt-gen/src/writer/rust/mod.rs:20]: Should this also contain the convert include?
``` rust
include!("cxx-qt-lib/include/convert.h");
```

**Review by LeonMatthesKDAB — CHANGES_REQUESTED** [review]
Going in the right direction, just a bit of nitpicking.

File: cxx-qt-gen/src/writer/rust/mod.rs (hunk 1 above)
Comment (ahayzen-kdab) [reply cxx-qt-gen/src/writer/rust/mod.rs]: Done

File: cxx-qt-gen/src/writer/rust/mod.rs (hunk 2 above)
Comment (ahayzen-kdab) [reply cxx-qt-gen/src/writer/rust/mod.rs:19]: Done

**Review by Be-ing — APPROVED** [review]

**Pull request merged**